make.js

```diff
@@ -333,17 +333,26 @@ target.test = function() {
     var pattern1 = buildPath + '/' + taskType + '/Tests/' + suiteType + '.js';
     var pattern2 = buildPath + '/Common/' + taskType + '/Tests/' + suiteType + '.js';
     var pattern3 = buildTestsPath + '/' + suiteType + '.js';
-    var testsSpec = matchFind(pattern1, buildPath)
-        .concat(matchFind(pattern2, buildPath))
-        .concat(matchFind(pattern3, buildTestsPath, { noRecurse: true }));
+    
+    var testsSpec = [];
+	
+    if (matchFind(pattern1, buildPath).length > 0) {
+	testsSpec.push(pattern1);
+    }
+    if (matchFind(pattern2, buildPath).length > 0) {
+	testsSpec.push(pattern2);
+    }
+	
+    testsSpec.concat(matchFind(pattern3, buildTestsPath, { noRecurse: true }));
```

Comment:
This one doesn't need the length check?

Reply:
No, because we're adding the patterns we find directly to testsSpec (there should be either 0 or 1 patterns, so we can do that without blowing up the size)